dashboard/client/src/components/MetadataSection/MetadataSection.tsx

```diff
@@ -92,6 +96,28 @@ export const MetadataContentField: React.FC<{
   const classes = useStyles();
   const [copyIconClicked, setCopyIconClicked] = useState<boolean>(false);
 
+  const copyElement = content && "copyableValue" in content && (
+    <Tooltip
+      placement="top"
+      title={copyIconClicked ? "Copied" : "Click to copy"}
+    >
+      <IconButton
+        aria-label="copy"
+        onClick={() => {
+          setCopyIconClicked(true);
+          copy(content.copyableValue);
+        }}
+        // Set up mouse events to avoid text changing while tooltip is visible
+        onMouseEnter={() => setCopyIconClicked(false)}
+        onMouseLeave={() => setTimeout(() => setCopyIconClicked(false), 333)}
+        size="small"
+        className={classes.button}
+      >
+        <RiFileCopyLine />
+      </IconButton>
+    </Tooltip>
+  );
+
   if (content === undefined || "value" in content) {
     return content === undefined || !("link" in content) ? (
```

Comment:
nit: Could we simplify the logic here? Like early catch content == undefined


if(content == undefined) return <...>

if ("value" in content) {
    .... 
}
if ("link" in content) {
    .... 
}